src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -664,12 +667,18 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
     };
   }, []);
 
-  const hasEnoughFundsForGas = useSharedValue<boolean | undefined>(undefined);
+  // Stop auto-fetching if there is a quote error or no input asset balance
   useAnimatedReaction(
-    () => isFetching.value,
-    fetching => {
-      if (fetching) hasEnoughFundsForGas.value = undefined;
-    }
+    () =>
+      SwapWarning.swapWarning.value.type === SwapWarningType.no_quote_available ||
+      SwapWarning.swapWarning.value.type === SwapWarningType.no_route_found ||
+      (internalSelectedInputAsset.value && equalWorklet(internalSelectedInputAsset.value.maxSwappableAmount, '0')),
+    (shouldStop, previous) => {
+      if (shouldStop && previous === false) {
+        SwapInputController.quoteFetchingInterval.stop();
+      }
+    },
+    []
```

Comment:
does this require cleanup for the areas we were previously calling `.stop()`?

Reply:
I guess technically calling .stop is a no-op in the case where it's already been stopped so nvm